internal/rego/rego.go

```diff
@@ -177,16 +180,36 @@ func (r Rego) PolicyID() string {
 // found in the header comment of the rego file.
 func (r Rego) Description() string {
 	var description string
+	var handlingCodeBlock bool
 	for _, comment := range r.headerComments {
-		if strings.HasPrefix(comment, "@") {
+		if strings.HasPrefix(strings.TrimSpace(comment), "@") {
 			continue
 		}
 
-		description += comment
+		// By default, we trim the comments found in the header to produce better looking documentation.
+		// However, when a comment in the Rego starts with a code block, we do not want to format
+		// any of the text within the code block.
+		if strings.HasPrefix(strings.TrimSpace(comment), "```") {
+
+			// Everytime we see a code block marker, we want to flip the status of whether or
+			// not we are currently handling a code block.
+			//
+			// i.e. The first time we see a codeblock marker we are handling a codeblock.
+			//      The second time we see a codeblock marker, we are no longer handling that codeblock.
+			handlingCodeBlock = !handlingCodeBlock
+		}
+
+		if handlingCodeBlock {
+			description += comment
+		} else {
+			description += strings.TrimSpace(comment)
+		}
```

Comment:
This section here is the core of the change. It feels straight-forward enough but am open to suggestions!